src/librustc_trans/mir/block.rs

```diff
@@ -284,6 +284,7 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
 
                 // After this point, bcx is the block for the call to panic.
                 bcx = panic_block.build();
+                debug_loc.apply_to_bcx(&bcx);
```

Comment:
When old trans is gone, we can have a simpler and less error-prone interface, i.e. a `Block` that always knows its debug position.